src/snowflake/connector/compute_chunk_size.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+constants = {
+    "CURRENT_CHUNK_SIZE": 8388608,
+    "MAX_OBJECT_SIZE": 5497558138880,
+    "MAX_PART_SIZE": 5368709120,
+    "MIN_PART_SIZE": 5242880,
+    "MAX_PARTS": 10000,
+}
+
+
+class ChunkSizeCalculator:
+    def __init__(
+        self,
+        current_chunk_size: int = constants["CURRENT_CHUNK_SIZE"],
+        max_object_size: int = constants["MAX_OBJECT_SIZE"],
+        max_part_size: int = constants["MAX_PART_SIZE"],
+        min_part_size: int = constants["MIN_PART_SIZE"],
+        max_parts: int = constants["MAX_PARTS"],
+    ) -> None:
+        self.current_chunk_size = current_chunk_size
+        self.max_object_size = max_object_size
+        self.max_part_size = max_part_size
+        self.min_part_size = min_part_size
+        self.max_parts = max_parts
+
+    def compute_chunk_size(self, file_size: int = None) -> int:
+        chunk_size = self.current_chunk_size
+        # check if we don"t exceed the allowed max file size 5 TiB
+        if file_size is not None and file_size < self.max_object_size:
```

Comment:
reading from the code, the file_size can not be None, otherwise we will simply throw an error.
in this case this parameter should be a normal positional parameter without default value being None

Reply:
Got rid of the default value